### PR TITLE
Fix the virtual destructor warning

### PR DESCRIPTION
**Pull request by janvorli**
This change changes destructors to be virtual or adds virtual ones where they were
missing based on the clang warnings.
At several places, objects were being deleted by invoking delete on an interface pointer, so I've fixed those.

### CODE REVIEW

**Comment by janvorli** [comment]
Fixes part of the issue #720

**Comment by janvorli** [comment]
@jkotas can you take a look please?

File: src/debug/daccess/daccess.cpp [hunk 1]
```diff
@@ -1334,7 +1334,7 @@ SplitName::CdNextField(ClrDataAccess* dac,
                                       fieldTypeHandle);
             if (!*fieldType && tokenScopeRet)
             {
-                delete *tokenScopeRet;
+                delete (ClrDataModule*)*tokenScopeRet;
```
Comment (jkotas) [inline src/debug/daccess/daccess.cpp]: There is identical code in src\debug\daccess\inspect.cpp. Fix it too?

**Comment by jkotas** [comment]
LGTM

**Pull request merged**